src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -391,6 +409,8 @@ default Course findWithEagerLearningPathsAndCompetenciesByIdElseThrow(long cours
         return findWithEagerLearningPathsAndCompetenciesById(courseId).orElseThrow(() -> new EntityNotFoundException("Course", courseId));
     }
 
+    Page<Course> findByTitleIgnoreCaseContaining(@Param("partialTitle") String partialTitle, Pageable pageable);
```

Comment:
Afaik `@Param` is only necessary in combination with `@Query`.
```suggestion
    Page<Course> findByTitleIgnoreCaseContaining(String partialTitle, Pageable pageable);
```